src/mlpack/bindings/go/print_go.cpp

```diff
@@ -0,0 +1,298 @@
+/**
+ * @file print_go.cpp
+ * @author Yasmine Dumouchel
+ *
+ * Implementation of function to generate a .go file given a list of parameters
+ * for the function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include "print_go.hpp"
+#include <mlpack/core/util/cli.hpp>
+#include <mlpack/core/util/hyphenate_string.hpp>
+#include <set>
+
+using namespace mlpack::util;
+using namespace std;
+
+namespace mlpack {
+namespace bindings {
+namespace go {
+
+/**
+ * Given a list of parameter definition and program documentation, print a
+ * generated .go file to stdout.
+ *
+ * @param parameters List of parameters the program will use (from CLI).
+ * @param programInfo Documentation for the program.
+ * @param functionName Name of the function (i.e. "pca").
+ */
+void PrintGo(const util::ProgramDoc& programInfo,
+              const std::string& functionName)
+{
+  // Restore parameters.
+  CLI::RestoreSettings(programInfo.programName);
+
+  const std::map<std::string, util::ParamData>& parameters = CLI::Parameters();
+  typedef std::map<std::string, util::ParamData>::const_iterator ParamIter;
+
+  // Split into input and output parameters.  Take two passes on the input
+  // parameters, so that we get the required ones first.
+  vector<string> inputOptions, outputOptions;
+  for (ParamIter it = parameters.begin(); it != parameters.end(); ++it)
+  {
+    const util::ParamData& d = it->second;
+    if (d.input && d.required)
+    {
+      // Ignore some parameters.
+      if (d.name != "help" && d.name != "info" && d.name != "version")
+        inputOptions.push_back(it->first);
+    }
+    else if (!d.input)
+    {
+      outputOptions.push_back(it->first);
+    }
+  }
+
+  for (ParamIter it = parameters.begin(); it != parameters.end(); ++it)
+  {
+    const util::ParamData& d = it->second;
+    if (d.input && !d.required &&
+        d.name != "help" && d.name != "info" &&
+        d.name != "version")
+      inputOptions.push_back(it->first);
+  }
+
+  // First, we must generate the mlpack package name.
+  cout << "package mlpack" << endl;
+  cout << endl;
+
+  // Now we must print the cgo's import libraries and files.
+  cout << "/*" << endl;
+  cout << "#cgo CFLAGS: -I/capi -I. -g -Wall" << endl;
```

Comment:
Should this be `-I./capi`?  Otherwise it would reference the `/capi` folder from the root of the filesystem.  Also we should probably remove `-g` at some point.